ProjectWar/GameException.h

```diff
@@ -33,5 +33,20 @@ class InvalidPositionException : public std::runtime_error {
     int width;
     int height;
 };
+        
+class IllegalStateException : public std::runtime_error {
+public:
+    IllegalStateException( std::string msg )
```

Comment:
It is better to pass const std::string& since it would just take 4 bytes to storage the reference instead of the whole string which is larger almost always.